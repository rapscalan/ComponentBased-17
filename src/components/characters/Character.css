.Character {
  margin: 10px;

  ul {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background: #333333;
    padding: 25px;
    border-radius: 0 0 15px 15px;
  }

  li {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    border-bottom: 1px solid rgb(68,68,68);
    padding: 15px 0px 7.5px;
  }

  .left {
    color: #9E9E9E;
  }

  .right {
    color: #FF9800;
    text-align: end;
  }

  img {
    width: 100%;
    height: auto;
    border-radius: 15px 15px 0 0;
  }

  .banner {
    width: 390px;
    opacity: 0.8;
    background: rgb(32, 35, 41);
    padding: 0.625rem;
    position: absolute;
    bottom: 0;
  }

  h2 {
    color: rgb(245, 245, 245);
    font-size: 1.625rem;
  }

  p {
    color: rgb(187, 187, 187);
    margin: 0px;
  }
}