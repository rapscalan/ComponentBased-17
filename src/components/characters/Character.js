import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

//const characters = require('../../data/rickandmorty.json');

class Character extends Component {
  static propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    origin: PropTypes.string.isRequired,
    gender: PropTypes.string.isRequired,
    lastLocation: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }
  // characters.forEach(element => {
  //   console.log(el.name);
  // });

  render() {
    const { id, name, status, species, origin, gender, lastLocation, image } = this.props;
    // characters.forEach(el => {
    //   console.log(el.name);
    // });

    return (

      <>
        <div className={styles.Character}>
          {/* <section> */}
            <div style={{ position: 'relative' }}>
              <img src={image} alt={name}/>
              <div><h2>{name}</h2></div>
            </div>
            <ul>
              <li>
                <span className={styles.left}>Status</span>
                <span className={styles.right}>{status}</span>
              </li>
              <li>
                <span className={styles.left}>Species</span>
                <span className={styles.right}>{species}</span>
              </li>
              <li>
                <span className={styles.left}>Gender</span>
                <span className={styles.right}>{gender}</span>
              </li>
              <li>
                <span className={styles.left}>Origin</span>
                <span className={styles.right}>{origin}</span>
              </li>
              <li>
                <span className={styles.left}>Last Location</span>
                <span className={styles.right}>{lastLocation}</span>
              </li>
            </ul>
          {/* </section> */}
        </div>
      </ >
    );
  }
}

export default Character;
