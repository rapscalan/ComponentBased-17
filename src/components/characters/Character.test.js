import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';
import rickImage from '../../assets/1.jpeg';

describe('Character component', () => {
  const chars = require('../../data/rickandmorty.json');
  it('renders Character', () => {
    const wrapper = shallow(<Character image={rickImage} name={chars[0].name} id={chars[0].id} status={chars[0].status} species={chars[0].species} gender={chars[0].gender} origin={chars[0].origin.name} lastLocation={chars[0].location.name}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
