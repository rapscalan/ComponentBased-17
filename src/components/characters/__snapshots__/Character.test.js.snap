// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<Fragment>
  <div
    className="Character"
  >
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <img
        alt="Rick Sanchez"
        src="/path/image"
      />
      <div>
        <h2>
          Rick Sanchez
        </h2>
      </div>
    </div>
    <ul>
      <li>
        <span
          className="left"
        >
          Status
        </span>
        <span
          className="right"
        >
          Alive
        </span>
      </li>
      <li>
        <span
          className="left"
        >
          Species
        </span>
        <span
          className="right"
        >
          Human
        </span>
      </li>
      <li>
        <span
          className="left"
        >
          Gender
        </span>
        <span
          className="right"
        >
          Male
        </span>
      </li>
      <li>
        <span
          className="left"
        >
          Origin
        </span>
        <span
          className="right"
        >
          Earth (C-137)
        </span>
      </li>
      <li>
        <span
          className="left"
        >
          Last Location
        </span>
        <span
          className="right"
        >
          Earth (Replacement Dimension)
        </span>
      </li>
    </ul>
  </div>
</Fragment>
`;
